File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-option-shared-5-4/Restructuring.cpp
// Restructuring.cpp 
#include "Restructuring.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Restructuring::Restructuring(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //applicableNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* applicableNode = xmlNode->FirstChildElement("applicable");

   if(applicableNode){applicableIsNull_ = false;}
   else{applicableIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- applicableNode , address : " << applicableNode << std::endl;
   #endif
   if(applicableNode)
   {
      if(applicableNode->Attribute("href") || applicableNode->Attribute("id"))
      {
          if(applicableNode->Attribute("id"))
          {
             applicableIDRef_ = applicableNode->Attribute("id");
             applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));
             applicable_->setID(applicableIDRef_);
             IDManager::instance().setID(applicableIDRef_,applicable_);
          }
          else if(applicableNode->Attribute("href")) { applicableIDRef_ = applicableNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { applicable_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(applicableNode));}
   }

   //restructuringTypeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* restructuringTypeNode = xmlNode->FirstChildElement("restructuringType");

   if(restructuringTypeNode){restructuringTypeIsNull_ = false;}
   else{restructuringTypeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- restructuringTypeNode , address : " << restructuringTypeNode << std::endl;
   #endif
   if(restructuringTypeNode)
   {
      if(restructuringTypeNode->Attribute("href") || restructuringTypeNode->Attribute("id"))
      {
          if(restructuringTypeNode->Attribute("id"))
          {
             restructuringTypeIDRef_ = restructuringTypeNode->Attribute("id");
             restructuringType_ = boost::shared_ptr<RestructuringType>(new RestructuringType(restructuringTypeNode));
             restructuringType_->setID(restructuringTypeIDRef_);
             IDManager::instance().setID(restructuringTypeIDRef_,restructuringType_);
          }
          else if(restructuringTypeNode->Attribute("href")) { restructuringTypeIDRef_ = restructuringTypeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { restructuringType_ = boost::shared_ptr<RestructuringType>(new RestructuringType(restructuringTypeNode));}
   }

   //multipleHolderObligationNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* multipleHolderObligationNode = xmlNode->FirstChildElement("multipleHolderObligation");

   if(multipleHolderObligationNode){multipleHolderObligationIsNull_ = false;}
   else{multipleHolderObligationIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- multipleHolderObligationNode , address : " << multipleHolderObligationNode << std::endl;
   #endif
   if(multipleHolderObligationNode)
   {
      if(multipleHolderObligationNode->Attribute("href") || multipleHolderObligationNode->Attribute("id"))
      {
          if(multipleHolderObligationNode->Attribute("id"))
          {
             multipleHolderObligationIDRef_ = multipleHolderObligationNode->Attribute("id");
             multipleHolderObligation_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(multipleHolderObligationNode));
             multipleHolderObligation_->setID(multipleHolderObligationIDRef_);
             IDManager::instance().setID(multipleHolderObligationIDRef_,multipleHolderObligation_);
          }
          else if(multipleHolderObligationNode->Attribute("href")) { multipleHolderObligationIDRef_ = multipleHolderObligationNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { multipleHolderObligation_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(multipleHolderObligationNode));}
   }

   //multipleCreditEventNoticesNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* multipleCreditEventNoticesNode = xmlNode->FirstChildElement("multipleCreditEventNotices");

   if(multipleCreditEventNoticesNode){multipleCreditEventNoticesIsNull_ = false;}
   else{multipleCreditEventNoticesIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- multipleCreditEventNoticesNode , address : " << multipleCreditEventNoticesNode << std::endl;
   #endif
   if(multipleCreditEventNoticesNode)
   {
      if(multipleCreditEventNoticesNode->Attribute("href") || multipleCreditEventNoticesNode->Attribute("id"))
      {
          if(multipleCreditEventNoticesNode->Attribute("id"))
          {
             multipleCreditEventNoticesIDRef_ = multipleCreditEventNoticesNode->Attribute("id");
             multipleCreditEventNotices_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(multipleCreditEventNoticesNode));
             multipleCreditEventNotices_->setID(multipleCreditEventNoticesIDRef_);
             IDManager::instance().setID(multipleCreditEventNoticesIDRef_,multipleCreditEventNotices_);
          }
          else if(multipleCreditEventNoticesNode->Attribute("href")) { multipleCreditEventNoticesIDRef_ = multipleCreditEventNoticesNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { multipleCreditEventNotices_ = boost::shared_ptr<XsdTypeBoolean>(new XsdTypeBoolean(multipleCreditEventNoticesNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeBoolean> Restructuring::getApplicable()
{
   if(!this->applicableIsNull_){
        if(applicableIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(applicableIDRef_));
        }else{
             return this->applicable_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<RestructuringType> Restructuring::getRestructuringType()
{
   if(!this->restructuringTypeIsNull_){
        if(restructuringTypeIDRef_ != ""){
             return boost::shared_static_cast<RestructuringType>(IDManager::instance().getID(restructuringTypeIDRef_));
        }else{
             return this->restructuringType_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<RestructuringType>();
   }
}
boost::shared_ptr<XsdTypeBoolean> Restructuring::getMultipleHolderObligation()
{
   if(!this->multipleHolderObligationIsNull_){
        if(multipleHolderObligationIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(multipleHolderObligationIDRef_));
        }else{
             return this->multipleHolderObligation_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
boost::shared_ptr<XsdTypeBoolean> Restructuring::getMultipleCreditEventNotices()
{
   if(!this->multipleCreditEventNoticesIsNull_){
        if(multipleCreditEventNoticesIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeBoolean>(IDManager::instance().getID(multipleCreditEventNoticesIDRef_));
        }else{
             return this->multipleCreditEventNotices_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeBoolean>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-option-shared-5-4/Restructuring.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Restructuring : ISerialized
    {
        public Restructuring(XmlNode xmlNode)
        {
            XmlNode applicableNode = xmlNode.SelectSingleNode("applicable");
            
            if (applicableNode != null)
            {
                if (applicableNode.Attributes["href"] != null || applicableNode.Attributes["id"] != null) 
                {
                    if (applicableNode.Attributes["id"] != null) 
                    {
                        applicableIDRef_ = applicableNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(applicableNode);
                        IDManager.SetID(applicableIDRef_, ob);
                    }
                    else if (applicableNode.Attributes["href"] != null)
                    {
                        applicableIDRef_ = applicableNode.Attributes["href"].Value;
                    }
                    else
                    {
                        applicable_ = new XsdTypeBoolean(applicableNode);
                    }
                }
                else
                {
                    applicable_ = new XsdTypeBoolean(applicableNode);
                }
            }
            
        
            XmlNode restructuringTypeNode = xmlNode.SelectSingleNode("restructuringType");
            
            if (restructuringTypeNode != null)
            {
                if (restructuringTypeNode.Attributes["href"] != null || restructuringTypeNode.Attributes["id"] != null) 
                {
                    if (restructuringTypeNode.Attributes["id"] != null) 
                    {
                        restructuringTypeIDRef_ = restructuringTypeNode.Attributes["id"].Value;
                        RestructuringType ob = new RestructuringType(restructuringTypeNode);
                        IDManager.SetID(restructuringTypeIDRef_, ob);
                    }
                    else if (restructuringTypeNode.Attributes["href"] != null)
                    {
                        restructuringTypeIDRef_ = restructuringTypeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        restructuringType_ = new RestructuringType(restructuringTypeNode);
                    }
                }
                else
                {
                    restructuringType_ = new RestructuringType(restructuringTypeNode);
                }
            }
            
        
            XmlNode multipleHolderObligationNode = xmlNode.SelectSingleNode("multipleHolderObligation");
            
            if (multipleHolderObligationNode != null)
            {
                if (multipleHolderObligationNode.Attributes["href"] != null || multipleHolderObligationNode.Attributes["id"] != null) 
                {
                    if (multipleHolderObligationNode.Attributes["id"] != null) 
                    {
                        multipleHolderObligationIDRef_ = multipleHolderObligationNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(multipleHolderObligationNode);
                        IDManager.SetID(multipleHolderObligationIDRef_, ob);
                    }
                    else if (multipleHolderObligationNode.Attributes["href"] != null)
                    {
                        multipleHolderObligationIDRef_ = multipleHolderObligationNode.Attributes["href"].Value;
                    }
                    else
                    {
                        multipleHolderObligation_ = new XsdTypeBoolean(multipleHolderObligationNode);
                    }
                }
                else
                {
                    multipleHolderObligation_ = new XsdTypeBoolean(multipleHolderObligationNode);
                }
            }
            
        
            XmlNode multipleCreditEventNoticesNode = xmlNode.SelectSingleNode("multipleCreditEventNotices");
            
            if (multipleCreditEventNoticesNode != null)
            {
                if (multipleCreditEventNoticesNode.Attributes["href"] != null || multipleCreditEventNoticesNode.Attributes["id"] != null) 
                {
                    if (multipleCreditEventNoticesNode.Attributes["id"] != null) 
                    {
                        multipleCreditEventNoticesIDRef_ = multipleCreditEventNoticesNode.Attributes["id"].Value;
                        XsdTypeBoolean ob = new XsdTypeBoolean(multipleCreditEventNoticesNode);
                        IDManager.SetID(multipleCreditEventNoticesIDRef_, ob);
                    }
                    else if (multipleCreditEventNoticesNode.Attributes["href"] != null)
                    {
                        multipleCreditEventNoticesIDRef_ = multipleCreditEventNoticesNode.Attributes["href"].Value;
                    }
                    else
                    {
                        multipleCreditEventNotices_ = new XsdTypeBoolean(multipleCreditEventNoticesNode);
                    }
                }
                else
                {
                    multipleCreditEventNotices_ = new XsdTypeBoolean(multipleCreditEventNoticesNode);
                }
            }
            
        
        }
        
    
        #region Applicable_
        private XsdTypeBoolean applicable_;
        public XsdTypeBoolean Applicable_
        {
            get
            {
                if (this.applicable_ != null)
                {
                    return this.applicable_; 
                }
                else if (this.applicableIDRef_ != null)
                {
                    applicable_ = IDManager.getID(applicableIDRef_) as XsdTypeBoolean;
                    return this.applicable_; 
                }
                else
                {
                      return this.applicable_; 
                }
            }
            set
            {
                if (this.applicable_ != value)
                {
                    this.applicable_ = value;
                }
            }
        }
        #endregion
        
        public string applicableIDRef_ { get; set; }
        #region RestructuringType_
        private RestructuringType restructuringType_;
        public RestructuringType RestructuringType_
        {
            get
            {
                if (this.restructuringType_ != null)
                {
                    return this.restructuringType_; 
                }
                else if (this.restructuringTypeIDRef_ != null)
                {
                    restructuringType_ = IDManager.getID(restructuringTypeIDRef_) as RestructuringType;
                    return this.restructuringType_; 
                }
                else
                {
                      return this.restructuringType_; 
                }
            }
            set
            {
                if (this.restructuringType_ != value)
                {
                    this.restructuringType_ = value;
                }
            }
        }
        #endregion
        
        public string restructuringTypeIDRef_ { get; set; }
        #region MultipleHolderObligation_
        private XsdTypeBoolean multipleHolderObligation_;
        public XsdTypeBoolean MultipleHolderObligation_
        {
            get
            {
                if (this.multipleHolderObligation_ != null)
                {
                    return this.multipleHolderObligation_; 
                }
                else if (this.multipleHolderObligationIDRef_ != null)
                {
                    multipleHolderObligation_ = IDManager.getID(multipleHolderObligationIDRef_) as XsdTypeBoolean;
                    return this.multipleHolderObligation_; 
                }
                else
                {
                      return this.multipleHolderObligation_; 
                }
            }
            set
            {
                if (this.multipleHolderObligation_ != value)
                {
                    this.multipleHolderObligation_ = value;
                }
            }
        }
        #endregion
        
        public string multipleHolderObligationIDRef_ { get; set; }
        #region MultipleCreditEventNotices_
        private XsdTypeBoolean multipleCreditEventNotices_;
        public XsdTypeBoolean MultipleCreditEventNotices_
        {
            get
            {
                if (this.multipleCreditEventNotices_ != null)
                {
                    return this.multipleCreditEventNotices_; 
                }
                else if (this.multipleCreditEventNoticesIDRef_ != null)
                {
                    multipleCreditEventNotices_ = IDManager.getID(multipleCreditEventNoticesIDRef_) as XsdTypeBoolean;
                    return this.multipleCreditEventNotices_; 
                }
                else
                {
                      return this.multipleCreditEventNotices_; 
                }
            }
            set
            {
                if (this.multipleCreditEventNotices_ != value)
                {
                    this.multipleCreditEventNotices_ = value;
                }
            }
        }
        #endregion
        
        public string multipleCreditEventNoticesIDRef_ { get; set; }
        
    
        
    
    }
    
}

